front/lib/vaults.ts

```diff
@@ -24,3 +32,13 @@ export const dustAppsListUrl = (
 ): string => {
   return `/w/${owner.sId}/vaults/${vault.sId}/categories/apps`;
 };
+
+export const groupVaults = (vaults: VaultType[]) => {
+  // Group by kind and sort.
+  const groupedVaults = groupBy(vaults, (vault) => vault.kind);
+  return VAULTS_SORT_ORDER.map((kind) => ({
+    kind,
+    vaults: groupedVaults[kind] || [],
+    // remove the empty system menu for users & builders
+  })).filter(({ vaults, kind }) => !(kind !== "system" && vaults.length === 0));
```

Comment:
Is the comment outdated here?

Reply:
Comment is still valid but the test was broken .. 🙈

Reply:
moving the filter to VaultNav